setuptools/command/bdist_egg.py

```diff
@@ -331,6 +332,8 @@ def walk_egg(egg_dir):
     """Walk an unpacked egg's contents, skipping the metadata directory"""
     walker = os.walk(egg_dir)
     base, dirs, files = next(walker)
+    dirs.sort()
```

Comment:
It's starting to feel like this change is getting sprinkled around a few places. Maybe it would make more sense to have a "sorted walk" function or similar.